whylog/front/whylog_factory.py

```diff
@@ -0,0 +1,45 @@
+import yaml
+
+from whylog.assistant.const import AssistantType
+from whylog.assistant.regex_assistant import RegexAssistant
+from whylog.config import YamlConfig
+from whylog.log_reader import LogReader
+from whylog.teacher import Teacher
+
+
+def _teachers_generator(config, pattern_assistant):
+    """
+    Every call of this function returns new teacher for new lesson.
+    """
+    while True:
+        yield Teacher(config, pattern_assistant)
+
+
+def create_pattern_assistant(config):
+    """
+    This function read from which pattern assistant should be created.
+    The default one now is RegexAssistant.
+    """
+    try:
+        if config['pattern_assistant'] == AssistantType.REGEX:
+            return RegexAssistant()
+    except KeyError:
```

Comment:
This function is wrong. Please go back to first phase of review and focus on what it does and on what it was supposed to do.